### PR TITLE
PDF Export API

### PR DESCRIPTION
**Pull request by hungnguyen94**
Added an API which returns a PDF download. Moved PDF Export to utility classes. 
Added AngularJS code to call the corresponding API and download button to let Spring Boot serve a download.

### CODE REVIEW

File: src/main/java/nl/tudelft/thefirstorder/web/rest/ProjectResource.java [hunk 1]
```diff
@@ -141,4 +144,41 @@
...
+        log.debug("Export Project {} to PDF.", id);
+        Project currentProject = project.get();
+        Resource resource = PDFExportUtil.exportProjectToPDF(currentProject);
+
+        return ResponseEntity.ok()
+                .headers(headers)
+                .contentLength(resource.contentLength())
+                .contentType(MediaType.parseMediaType("application/octet-stream"))
+                .body(resource);
+    }
```
Comment (rubenwiersma) [inline src/main/java/nl/tudelft/thefirstorder/web/rest/ProjectResource.java]: Is this protected from unauthorized users?

**Comment by rubenwiersma** [comment]
Looking good, we've seen it tested.

**Comment by cadw** [comment]
Looks mergeable to me!

**Comment by trombomart** [comment]
@rubenwiersma the test for the pdfexport is in a different branch because this class was already merged to the dev before.

**Comment by rubenwiersma** [comment]
Okay, we'll merge it then.

File: src/main/java/nl/tudelft/thefirstorder/web/rest/ProjectResource.java (hunk 1 above)
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/web/rest/ProjectResource.java]: Yes, it requires an authorization token to be send with call. You won't be able to call this API without being authorized.

**Pull request merged**